client-core/src/wallet/syncer.rs

```diff
@@ -565,9 +576,38 @@ impl<
             }
 
             // Fetch batch details if it cannot be fast forwarded
-            let blocks = self.env.client.block_batch(range.iter())?;
-            let block_results = self.env.client.block_results_batch(range.iter())?;
-            let states = self.env.client.query_state_batch(range.iter().cloned())?;
+            let mut blocks: Vec<Block>;
+            let mut block_results: Vec<BlockResultsResponse>;
+            let mut states: Vec<ChainState>;
+            // if any error occurs, do it again
```

Comment:
should there be some limited number of retries?

Reply:
i'll add some limited number for retries